test/modules/generic-mqtt-tester/src/tester.rs

```diff
@@ -177,13 +221,23 @@ impl MessageTester {
         match exited {
             Err(e) => {
                 error!("Stopping test run because task exited with error: {:?}", e);
-                self.shutdown_handle.shutdown().await?;
+                self.shutdown_handle
+                    .shutdown(ExitedWork::NoneOrUnknown)
+                    .await;
             }
             Ok(Err(e)) => {
                 error!("Stopping test run because task exited with error: {:?}", e);
-                self.shutdown_handle.shutdown().await?;
+                self.shutdown_handle
```

Comment:
What's the difference between these cases? Why do we sometimes get Err(e) and sometimes Ok(Err(e))

Reply:
and as a nit: I think you can do something like "Err(e) | Ok(Err(e))" and shove the duplicated logic together

Reply:
We push join handles into a collection and then select all on them. The outer error is a JoinHandle defined behavior for if the task panics. The inner error is because the spawned task has a return type of Result.

Reply:
Also, consolidating with this syntax didn't work for me because the errors are different types.